packages/enketo-core/src/js/form-model.js

```diff
@@ -389,6 +389,9 @@ FormModel.prototype.mergeXml = function (recordStr) {
         templateEls[i].remove();
     }
 
+    // Fix missing repeat nodes that may break editing
+    this.fixRecordRepeatData(record);
```

Comment:
nit: i'd rather not make another method that used only once but keep it here, separated by a multiple comment block like the ones above